script.js

```diff
@@ -154,48 +147,136 @@ var testArray = [1, 2, 4, 8, 16, 32, 64, 128];
+
+// 1. Prompt the user for a name, and show
+// (in any way, e.g. console.log) the phone number corresponding to
+// the given name.
+
+// var Phonebook = {
+//     "Jason"         :   91216599,
+//     "Amelie"        :   64239839,
+//     "Sing An"       :   85404834,
+//     "Ong Tng Kai"   :   94547788,
+//     "Nelson Tan"    :   63396565
+// };
+
+// var inputName = prompt("Enter a name");
+
+// var phonebookName = Object.keys(Phonebook);
+
+// // console.log(phonebookName); // ["Jason", "Amelie", "Sing An", "Ong Tng Kai", "Nelson Tan"]
+
+// for (var i = 0; i < phonebookName.length; i++) {
+//   if (inputName == phonebookName[i]) {
+//     console.log("The phone number for " + phonebookName[i] + " is: " + Phonebook[inputName])
+//   }
+// }
+
+// ====================================================================
+// Write your solution 2 below.
 // ====================================================================
 
-var Phonebook = {
-    "Jason"         :   91216599,
-    "Amelie"        :   64239839,
-    "Sing An"       :   85404834,
-    "Ong Tng Kai"   :   94547788,
-    "Nelson Tan"    :   63396565
-};
+// 2. Prompt the user for a name and a number. Update the number
+// corresponding to the name if the name exists in the phonebook.
+
+// var Phonebook = {
+//     "Jason"         :   91216599,
+//     "Amelie"        :   64239839,
+//     "Sing An"       :   85404834,
+//     "Ong Tng Kai"   :   94547788,
+//     "Nelson Tan"    :   63396565
+// };
 
+// var updatePhonebookName = prompt("Enter a name");
 
+// var updatePhonebookNumber = parseInt(prompt("Enter a number"));
 
+// var phonebookKeys = Object.keys(Phonebook);
 
+// // var eachKey = phonebookKeys.forEach();
 
+// for (var i = 0; i < phonebookKeys.length; i++) {
+//   if (updatePhonebookName == phonebookKeys[i]) {
+//     Phonebook[updatePhonebookName] = updatePhonebookNumber;
+//     console.log("The number has been updated!");
+//   }
+// }
 
+// console.log(Phonebook);
 
+// ====================================================================
+// Write your solution 3 below.
+// ====================================================================
 
+// 3. Prompt the user for a name and a number. If the name and number
+// do not already exist in the phonebook, insert it into the phonebook.
 
+// var Phonebook = {
+//     "Jason"         :   91216599,
+//     "Amelie"        :   64239839,
+//     "Sing An"       :   85404834,
+//     "Ong Tng Kai"   :   94547788,
+//     "Nelson Tan"    :   63396565
+// };
 
+// var addPhonebookName = prompt("Enter a name");
 
+// var addPhonebookNumber = parseInt(prompt("Enter a number"));
 
+// var phonebookKeys = Object.keys(Phonebook);
 
+// // var eachKey = phonebookKeys.forEach();
 
+// for (var i = 0; i < phonebookKeys.length; i++) {
+//   if (addPhonebookName !== phonebookKeys[i]) {
+//     Phonebook[addPhonebookName] = addPhonebookNumber;
+//     console.log("The number has been added!");
+//   }
+// }
```

Comment:
This loop actually inserts the entry into the phonebook multiple times since the comparison is nearly always false. Luckily Phonebook[addPhonebookName] always refers to the same entry so we don't get multiple entries...